client/app/scripts/charts/nodes-layout.js

```diff
@@ -487,7 +487,7 @@ export function doLayout(immNodes, immEdges, opts) {
 
     // cache results
     cache.cachedLayout = layout;
-    cache.nodeCache = cache.nodeCache.merge(layout.nodes);
+    cache.nodeCache = cache.nodeCache.merge(layout.nodes.map(n => fromJS({ x: n.get('x'), y: n.get('y') })));
```

Comment:
It's completely up to you - I just wanted to make this suggestion alongside your current solution :)